core/src/c_api/tiledb_utils.cc

```
@@ -324,7 +324,7 @@ static int check_file_for_read(TileDB_CTX *tiledb_ctx, std::string filename) {
 }
 
 
-#include <openssl/md5.h>
+#include "tiledb_openssl_shim.h"
```

tiledb_openssl_shim.h has been included, but there seem to be no openssl3 related changes in this file. Why?

---

At line number 330, we are calling a function called MD5(), which is defined as weak attribute in tiledb_openssl_shim.h

---

If you are not changing anything, why not leave #include <openssl/md5.h> as is. If keeping it as is putting out deprecated warnings with openssl3, you will have to have implement the alternate md5 methods for openssl3 as you have done for aws and s3 to use tiledb_openssl_shim.h. Does that make sense?

---

I tried using the old file and getting the warnings like below.
So defined it as weak attribute in tiledb_openssl_shim.h & included it.

/nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/c_api/tiledb_utils.cc: In function ‘void TileDBUtils::print_md5_hash(unsigned char*, size_t)’:
/nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/c_api/tiledb_utils.cc:331:6: warning: ‘unsigned char* TileDBUtils::MD5(const unsigned char*, size_t, unsigned char*)’ is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  331 |   MD5(buffer, length, md);
      |   ~~~^~~~~~~~~~~~~~~~~~~~
In file included from /nfs/home/santhosh/workbenchKms/2tdb_ossl3_aws_gcs/TileDB/core/src/c_api/tiledb_utils.cc:328:
/usr/include/openssl/md5.h:52:38: note: declared here
   52 | OSSL_DEPRECATEDIN_3_0 unsigned char *MD5(const unsigned char *d, size_t n,
      |                                      ^~~

---

One of the aims of this exercise was to not just suppress deprecation warnings but to start using the openssl3 functionality instead of openssl1. Using `tiledb_openssl_shim` only suppresses the error, we should also be using the appropriate MD5 api from openssl3. See https://www.openssl.org/docs/man3.1/man3/MD5.html for what should be used for replacements.